### DISCUSSION TITLE
Welche Einstellungen für Tibber Sensor?

### DISCUSSION THREAD

**Discussion by Hutzelbrot** [hoylabs/OpenDTU-OnBattery#1237] [Q&A] [answered]
Die aktuelle Version unterstützt den Tibber Senor als Leistungsmesser. Aber wo und wie kann man das aktivieren?

**Comment by Sorbat** [comment, marked as answer]
meinst du das bei Stromzähler?
<img width="666" alt="Bildschirmfoto 2024-09-07 um 15 54 31" src="https://github.com/user-attachments/assets/48d13bda-9cc8-4d13-bd0b-59ec51b091dd">

**Comment by schlimmchen** [comment]
Ich denke, dass danach gefragt ist.

Im [Wiki](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Power-Meter#tibber-pulse-via-tibber-bridge) steht etwas dazu.

**Reply by Hutzelbrot** [reply]
Habe ich im Menue Stromzähler total übersehen. Und im Wiki auch. Oh jeh, das wird immer schlimmer. Danke!

**Comment by johannesosterberg** [comment]
Super, dass ihr jetzt Tibber mit einbezogen habt! Leider schreibt Tibber die Rohdaten des Stromzählers direkt in das "json", welches aus data.json?node_id=1 kommt. Bei mir sieht das dann folgendermaßen aus. Ich glaube, dieses Format nennt sich OBIS/ASCII.

```/xxx
1-0:0.0.0*255(xxx)
1-0:96.1.0*255(xxx)
1-0:1.8.0*255(004803.19690591*kWh)
1-0:2.8.0*255(000107.30434053*kWh)
1-0:16.7.0*255(000009.22*W)
1-0:36.7.0*255(000069.93*W)
1-0:56.7.0*255(-000123.86*W)
1-0:76.7.0*255(000063.15*W)
1-0:32.7.0*255(237.2*V)
1-0:52.7.0*255(237.1*V)
1-0:72.7.0*255(237.1*V)
1-0:96.5.0*255(001C2104)
0-0:96.8.0*255(0398ACAC)
!
```

Ich habe für den Hausgebrauch einen kleinen Code geschrieben, der seit 5 Monaten problemlos läuft. Vielleicht hilft es dem ein oder anderen.

```c++
String extractValueObisAscii(const char* data)
{
    const char* searchFor = "1-0:16.7.0*255(";
    const char* startPos = strstr(data, searchFor);

    if (startPos) {
        startPos += strlen(searchFor); // Bewege den Zeiger an das Ende der Suchzeichenfolge
        const char* endPos = strchr(startPos, ')');

        if (endPos) {
            // Berechne die Länge des extrahierten Wertes
            int length = endPos - startPos;

            // Erstelle einen Buffer für den extrahierten Wert und füge das Nullzeichen hinzu
            char extractedValue[length + 1];
            strncpy(extractedValue, startPos, length);
            extractedValue[length] = '\0';

            // Konvertiere den C-String in einen Arduino-String und gib ihn zurück
            return String(extractedValue);
        }
    }
    // Gib einen leeren String zurück, wenn der Suchstring oder die schließende Klammer nicht gefunden wurde
    return "";
}
```

**Reply by schlimmchen** [reply]
Warum sehen die Daten an dieser URL für dich anders aus, also warum ist es kein SML? Kommt das aus einer Tibber Bridge? Nutzer, die das Feature nutzen, erhalten offensichtlich SML an dieser URL der Bridge. Hast du dazu was eingestellt?

**Reply by johannesosterberg** [reply]
Nein ich habe da nichts anderes eingestellt, das ist das was mein Stromzähler (eBZ DD3 Stadtwerke Kassel) ausgibt, es gibt noch ein paar andere Varianten, Tibber konvertiert das dann für sich, aber die /data.json?node_id=1 URL gibt einfach nur die Originaldaten des Zählers aus, bei mir eben OBIS in ASCI und nicht in HEX

**Reply by schlimmchen** [reply]
Da kriegt man ja Schnappatmung... Warum muss der Zähler das unbedingt anders formatiert rausblinken?!

Ich mach ein Issue draus und versuch es einzubauen.

Das heißt ja, dass auch ein Hichi die Daten so ausspuckt?! Kommt der denn damit klar? Also die Variante mit Tasmota und Wifi? Die TTL Variante des Hichi spuckt ja nur die Flanken des Lichtchens des Zählers aus... Das müsste dann wiederum OpenDTU-OnBattery erkennen und interpretieren, wie bei der Tibber Bridge auch.

Weißt du, wie sich das verhält?

**Reply by spcqike** [reply]
Wie werden die hichi denn abgefragt? Arbeiten die nicht mit tasmota? Do dass man die Daten per MQTT bekommt oder das tasmota json des Zählers abfragt? Tasmota kennt sowohl Obis als auch SML, man muss es im Tasmota Skript seines Zählers entsprechend konfigurieren. Das json was Tasmota macht ist dann aber immer gleich. 

https://tasmota.github.io/docs/Smart-Meter-Interface/#general-description

**Reply by johannesosterberg** [reply]
> Wie werden die hichi denn abgefragt? Arbeiten die nicht mit tasmota? Do dass man die Daten per MQTT bekommt oder das tasmota json des Zählers abfragt? Tasmota kennt sowohl Obis als auch SML, man muss es im Tasmota Skript seines Zählers entsprechend konfigurieren. Das json was Tasmota macht ist dann aber immer gleich.

 https://tasmota.github.io/docs/Smart-Meter-Interface/#general-description

Genau, ich konnte einfach OBIS ASCII einstellen und dann hat es funktioniert mittel MQTT. Aber als ich dann auf den Tibber-Tarif umgestellt habe, hat es leider nicht mehr funktioniert und ich habe mir das selbst zusammengebastelt.

**Reply by schlimmchen** [reply]
Roger, danke für die Infos und die Codevorlage. Ich mach demnächst ein Issue drauß und dann tracken wir das.

**Reply by johannesosterberg** [reply]
Cool, danke, vielleicht gibt es elegantere Codes, aber ich habe mir nicht stundenlang den Kopf darüber zerbrochen, ist auch nah am Arduino-Beispiel :)

**Reply by samluxh** [reply]
![222](https://github.com/user-attachments/assets/ffa731b2-e559-4c38-8d36-2e597e30d942)
![111](https://github.com/user-attachments/assets/075bb8c2-cfcf-488f-ae52-1a370bd51375)
@schlimmchen  Hallo, wurde dieses Issue gelöst?

Unter "http://{IP}/data.json?node_id=1" wird bei mir auch keine SML, sondern Text in dem Format wie oben (OBIS/ASCII?) gezeigt. 
Wenn ich nun auf den Button "Send HTTP(S) request und process response" klicke kommt immer "Result: 0.00W". 
Unter Live Data steht bei Grid Power entsprechend 0.0W, was natürlich nicht passt, Tibber App oder der Link selbst zeigt korrekte Werte an. 

Link zeigt das an:

/ABC
1-0:0.0.0*255(abc)
1-0:96.1.0*255(abc)
1-0:1.8.0*255(005314.30746024*kWh)
1-0:2.8.0*255(000008.36769750*kWh)
1-0:16.7.0*255(000695.56*W)
1-0:36.7.0*255(000655.54*W)
1-0:56.7.0*255(000040.02*W)
1-0:76.7.0*255(000000.00*W)
1-0:32.7.0*255(236.1*V)
1-0:52.7.0*255(238.3*V)
1-0:72.7.0*255(237.5*V)
1-0:96.5.0*255(001C0104)
0-0:96.8.0*255(037E26A6)
!

Es macht keinen Unterschied ob der Wert für 1-0:16.7.0*255 positiv oder negativ ist. 

In den Logs konnte ich nicht rausfinden, was genau ankommt, also warum da 0W angezeigt wird, da steht nur:
"2025-05-20T15:11:06.137+02:00 OpenDTU-FAE234 HOST=OpenDTU-FAE234 HOST_FROM=OpenDTU-FAE234 MESSAGE=Success MSGFORMAT=rfc5424 PID=xyz PROGRAM=OpenDTU SOURCE=s_network_udp TRANSPORT=rfc5426", falls dass der Log für diesen Prozess überhaupt ist, bin mir nicht sicher.

Eine Auswahl  zwischen SML oder OBIS wäre eventuell hilfreich, um den Fehler hier beim Text auszuschließen. Es wär für eine Fehlersuche auf jeden Fall auch nicht schlecht, wenn man dann auch sehen könnte was ankommt (z.B. die ganze OBIS Ausgabe).

Sucht sich openDTU-OnBattery dann automatisch die Stelle mit "1-0:16.7.0*255" raus, oder wie funktioniert es?
Wäre es nicht praktisch, wenn man selbst auswählen könnte welche Wert er nehmen/anzeigen soll (standardmäßig 16.7.0) und welche weiteren Werte auch sichtbar sein sollen?

Wiki hab ich durch, also Authorisierung als "Basic" gewählt und Username "admin". Das Passwort zum Gegentesten mal falsch eingegeben, dann kommt: "Bad HTTP code: 401". Beim Username ist es irgendwie egal was da steht, hauptsache das Password stimmt. Alle Kombinationen kurz getestet.

Würde mich freuen, wenn weitergeholfen werden kann. Lieben Dank schonmal! :-)

**Reply by samluxh** [reply]
@schlimmchen Gibt es evtl. News dazu? :-)

**Comment by anwista** [comment]
Hallo Gemeinde,

bin ganz neu hier und blick auch noch nicht so ganz durch mit MQTT usw.

Der Tibber ist mit dem Stromzähler und der Bridge verbunden und sendet auf die Tibber-App den Verbrauch in Echtzeit. Soweit funktioniert es. Die Bridge wird auch in der FritzBox im Wlan-Funknetz mit einer eigenen IP angezeigt.

Jetzt wollte ich auch den Tibber Pulse in die OpenDTU-OnBattery einbinden. Unter Stromzähler habe ich den Tibber ausgewählt, die URL "http://192.168.188...../data.json?node_id=1", unter Benutzer admin und unter Passwort den Code auf der Bridge (2x 4er Code mit Bindestrich) eingegeben. Leider kommt beim Konfiguration testen immer die Meldung "HTTP Error: connection refused.

Ich habe mir auch die Anleitung in der Wiki durchgelesen. Unter 5. sollte ich doch die http://10.133.70.1/params/ eingeben, aber die Seite ist nicht erreichbar. Oder muss ich da die IP eingeben, die im Wlan-Funknetz angezeigt wird (192.188.168.....? Die geht aber auch nicht. Stromzähler hab ich einen LandisGyr_E320, sollte also SML sein.

Mach ich irgendwas falsch oder kann mir jemand das erklären, so dass es auch ein Anfänger versteht?

vg
Andreas

**Reply by gitisgreat2023** [reply]
[Enabling web interface](https://github.com/marq24/ha-tibber-pulse-local) noch nicht gemacht?

**Reply by anwista** [reply]
nein, das kannte ich noch gar nicht, wird ja auch nirgends darauf hingewiesen.

Ich probiere ich mal aus, hoffe ich bekomm das hin

Erst einmal vielen Dank für den Hinweis 👍

**Reply by anwista** [reply]
Super, ich hab´s geschafft...

bis ich in den AccessPoint-Modus gekommen bin, musste ich etliche Male die Bridge ein- und ausstecken, aber irgendwann ging die Farbe endlich auf grün. Der Rest ging dann über die console wie beschrieben.

Jetzt habe ich die Netzanzeige im OpenDTU

vielen vielen Dank an gitisgreat2023

p.s. könnte man das Ganze nicht in der Wiki verlinken oder gleich reinschreiben? Ohne die Antworten hier im Forum wäre ich nie drauf gekommen..

**Reply by schlimmchen** [reply]
Du meinst [hier im Wiki](https://github.com/hoylabs/OpenDTU-OnBattery/wiki/Power-Meter#tibber-pulse-via-tibber-bridge)?

**Reply by anwista** [reply]
stimmt, da stand es ja auch, hab ich irgendwie überlesen... mein Englisch ist anscheinend etwas eingerostet ;-)

**Comment by samluxh** [comment]
Bekomme mit Tibber Pulse in der Live Ansicht von opendtuonbattery immer 0w angezeigt, ist einer auch von diesem Bug betroffen?

**Reply by Sorbat** [reply]
Bei mir funktionierte der, bis ich den Pulse durch einen anderen Powermeter ersetzt hatte. 

Funktionierte der Pulse schon einmal bei dir? Oder hast du ihn jetzt neu hinzugefügt und bekommst ihn nicht eingerichtet?

**Reply by samluxh** [reply]
Pulse an sich funktioniert ohne Probleme (läuft seit einigen Monaten), also die Werte werden korrekt in der Tibber App oder sonst wo (Homeassistant) angezeigt, nur in opendtuonbaterry bekomme ich nur 0w angezeigt.

Habe es oben etwas genauer beschrieben und bisher keine Rückmeldung bekommen:
https://github.com/hoylabs/OpenDTU-OnBattery/discussions/1237#discussioncomment-13208140

**Reply by anwista** [reply]
den web-server hattest du ja schon wie in der Beschreibung angegeben unter /params aktiviert, sonst würdest du ja nicht mit HA keine Werte angezeigt bekommen. Aber kommst jetzt über die zugeteilte IP auch über den browser drauf? Dann kann es ja nur noch an OpenDTU liegen. Hast Du die Firmware 2025.04.17 bei OpenDT-OnBattery drauf? Mit der läuft es bei mir, aber es gibt ja anscheinend aktuell noch eine Neuere.

**Reply by samluxh** [reply]
Per Browser bekomme ich über den Link "http://{IP}/data.json?node_id=1" halt die Werte von Tibber korrekt angezeigt, diese müssen nur noch in opendtuonbattery korrekt übernommen werden. Vielleicht liegts am OBIS/ASCII vs json Format.

Neueste Version (2025.06.11) wurde heute aufgespielt, Problem besteht immernoch.

**Reply by anwista** [reply]
bekommst du in deinem WLAN den tipper unter einer IP-Adresse angezeigt? Wenn ja, gib die mal mit http://192.168... in deinem Browser ein, dann solltest du auch die tibber-bridge angezeigt bekommen. Wenn nicht, dann hakt es irgendwo mit der WLAN-Verbindung.

**Reply by samluxh** [reply]
Per IP Adresse komme ich natürlich über den Browser auf Tibber Bridge, da gibts keine Probleme.

**Comment by Wingertsau** [comment]
Ich habe das gleiche Problem: 

- Web Server aktiviert wie im Wiki beschrieben (Das Aktivieren des lokalen WLANs der Bridge bedurfte einiger Versuche, klappte aber schliesslich; Was allerdings nicht im Wiki stand war folgendes : "With a recent firmware release the webserver_force_enable flag (39) does not appear any longer in the param list. So the console tab is the way to go! "  - Ich folgte den Anleitungen im [Link](https://github.com/marq24/ha-tibber-pulse-local?tab=readme-ov-file#via-console-tab) und gab über die Console die notwendige Info ein- Voila!)
-  Via IP-Adresse ( war ebenfalls ein wenig schwierig herauszufinden, ein Scan des Netzwerkes führte zu einer IP-Adresse mit folgender Vendor Bezeichnung: "Wistron Neweb Corporation") war dann die komplette Info wie oben von samluxh beschrieben vorhanden.
- Allerdings führt der Test bei Stromzähler HTTP(S) + SML - Konfiguration leider auch nur zu "Result: 0.00W"

Jetzt steht unter dieser Diskussion "Answered" die auf die eigentlich Frage zwar zutrifft, allerdings scheint hier ein weiteres neues Problem evtl. zusammenhängend mit neueren Firmwares auf der Bridge  aufzutreten?

**Reply by samluxh** [reply]
Hoffentlich wird das gefixed @schlimmchen 🙃

**Comment by Wingertsau** [comment]
Wäre bereit zu helfen, allerdings sind meine Programmierkenntnisse recht gering. Würde als Tester auch bereitstehen.
Könnte auch den Wikiartikel erweitern, weiß aber nicht ob ich die Rechte habe

**Comment by mlohnert** [comment]
Hallo,

hat das Problem mit  dem Tibber "Result: 0.00W" jemand gelöst? Bei mir läuft [2025.10.11](https://github.com/hoylabs/OpenDTU-OnBattery/releases/tag/2025.10.11). Die Web abfrage funktioniert, es gibt aber keine Spannungen in der Antwort.
/EBZ5DD32R06ETA_107
1-0:0.0.0*255(1EBZ0102025275)
1-0:96.1.0*255(1EBZ0102025275)
1-0:1.8.0*255(014415.66114775*kWh)
1-0:2.8.0*255(000490.50896484*kWh)
1-0:16.7.0*255(000220.75*W)
1-0:36.7.0*255(000171.70*W)
1-0:56.7.0*255(000379.77*W)
1-0:76.7.0*255(-000330.72*W)
1-0:96.5.0*255(001C4104)
0-0:96.8.0*255(0464CEA4)
!
In openDTUonB steht aber Result: 0.00W.
gruß

**Reply by samluxh** [reply]
Leider immernoch nicht gelöst...

**Comment by mlohnert** [comment]
Hallo,
hat jemand eine openDTUonB Entwicklungsumgebung as VM oder docker am Laufen? Ich hätte etwas Zeit über Weihnachten und da ich alle Komponenten hier am Laufen habe konnte ich da mal reinschauen. 

gruss

**Comment by DanielR92** [comment]
Achtung, es gab ein Update bei Tibber Gateway und da hat sich einiges geändert.

statt "metrics.json?node_id=1" jetzt "node_data.json?node_id=1"

**Comment by tonaltus** [comment]
bei mir war vorher nicht der ABfrage Code String 
http://xxx.xxx.xxx.xxx/netrics.json?node_id=1, sondern:
http://xxx.xxx.xxx.xxx/data.json?node_id=1

"data" habe ich nun geändert zu "node_data" und damit funktioniert die Abfrage tatsächlich wieder:

http://xxx.xxx.xxx.xxx/node_data.json?node_id=1

Vielen Dank, Daniel.
